cpp/src/centrality/betweenness_centrality_impl.cuh

```diff
@@ -548,27 +548,32 @@ rmm::device_uvector<weight_t> betweenness_centrality(
   std::optional<weight_t> scale_factor{std::nullopt};
 
   if (normalized) {
-    weight_t n = static_cast<weight_t>(graph_view.number_of_vertices());
-    if (!include_endpoints) { n -= weight_t{1}; }
-
-    scale_factor = n * (n - 1);
-  } else if (graph_view.is_symmetric())
+    if (include_endpoints) {
+      if (graph_view.number_of_vertices() >= 2) {
+        scale_factor = static_cast<weight_t>(
+          std::min(static_cast<vertex_t>(num_sources), graph_view.number_of_vertices()) *
+          (graph_view.number_of_vertices() - 1));
+      }
+    } else if (graph_view.number_of_vertices() > 2) {
+      scale_factor = static_cast<weight_t>(
+        std::min(static_cast<vertex_t>(num_sources), graph_view.number_of_vertices() - 1) *
+        (graph_view.number_of_vertices() - 2));
+    }
+  } else if (num_sources < static_cast<size_t>(graph_view.number_of_vertices())) {
+    scale_factor = (graph_view.is_symmetric() ? weight_t{2} : weight_t{1}) *
+                   static_cast<weight_t>(num_sources) /
+                   (include_endpoints ? static_cast<weight_t>(graph_view.number_of_vertices())
+                                      : static_cast<weight_t>(graph_view.number_of_vertices() - 1));
```

Comment:
We don't check `vertices.size()` (or sum of `vertices`.size() in multi-GPU) > 0. So, it is technically possible to pass empty seed vertices, and in this case, num_sources = 0 && graph_view.number_of_verties() = 1 is possible; then, scale_factor can become 0 leading to divide by 0.